### PR TITLE
Add `space_group_symmetry_operations_xyz` and sub-definition for sym ops

### PR DESCRIPTION
**Pull request by ml-evs**
Currently uses the proposed ECMA regexp at #488

Schemas will fail to generate unless we add `pattern` https://github.com/Materials-Consortia/OPTIMADE/pull/460#discussion_r1435595739 in #460

### CODE REVIEW

File: schemas/src/defs/v1.2/properties/optimade/common/symmetry_operation_xyz.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  **Requirements/Conventions:**
+
+  - The symmetry operation is described by a string that gives that symmetry operation in Jones' faithful representation (Bradley & Cracknell, 1972: pp. 35-37), adapted for computer string notation.
+  - The letters `x`, `y` and `z` that are typesetted with overbars in printed text represent coordinate values multiplied by -1 and are encoded as `-x`, `-y` and `-z`, respectively.
+  - The syntax of the strings representing symmetry operations MUST conform to regular expressions given in appendix `The Symmetry Operation String Regular Expressions`_.
+x-optimade-unit: inapplicable
+type:
+  - "string"
+maxLength: 3
+pattern: "^([-+]?[xyz]([-+][xyz])?([-+](1/2|[12]/3|[1-3]/4|[1-5]/6))?|[-+]?(1/2|[12]/3|[1-3]/4|[1-5]/6)([-+][xyz]([-+][xyz])?)?),([-+]?[xyz]([-+][xyz])?([-+](1/2|[12]/3|[1-3]/4|[1-5]/6))?|[-+]?(1/2|[12]/3|[1-3]/4|[1-5]/6)([-+][xyz]([-+][xyz])?)?),([-+]?[xyz]([-+][xyz])?([-+](1/2|[12]/3|[1-3]/4|[1-5]/6))?|[-+]?(1/2|[12]/3|[1-3]/4|[1-5]/6)([-+][xyz]([-+][xyz])?)?)$"
```
Comment (ml-evs) [inline schemas/src/defs/v1.2/properties/optimade/common/symmetry_operation_xyz.yaml]:
```suggestion
```

**Comment by ml-evs** [comment]
This is currently blocked by https://github.com/Materials-Consortia/OPTIMADE/pull/486

**Comment by rartino** [comment]
@ml-evs So #486 is merged, so that part is fine.

I know things have gotten a bit messy with the two PRs, and perhaps I should just merge #460 into this one and close #460, now that the two are in sync. But, in any case, this PR (as in #445) needs the meta-schema updated for the pattern field. But, ok, I'll merge this, and can add that.

**Comment by rartino** [comment]
@ml-evs But, eh, wait, how completely does this PR cover #464? I thought we at least modified the definition of some of the other symmetry fields, but maybe we didn't(?)

**Comment by ml-evs** [comment]
Hi @rartino, indeed this needed to be updated with the results of that PR (and also to include the other fields). I'll try to get around to that soon.

**Comment by rartino** [comment]
@ml-evs The changes relating to the other fields seemed quite minor, so in an effort to get things ready while I have time, I've added/modified them myself. I've also modified the meta schema for property definitions to allow the 'pattern' field for strings (but when/if #490 is merged, we should adjust the description). However, I though it to be fair to leave it to you add the fancy regex you worked out. Note: the place to add it is now in: `schemas/src/defs/v1.2/properties/optimade/common/space_group_symmetry_operation_xyz.yaml` which is the definition of a *single* symop, which is inherited by `space_group_symmetry_operations_xyz.yaml`.

**Pull request merged**